app/lib/search/handlers.dart

```diff
@@ -49,3 +52,17 @@ Future<shelf.Response> searchHandler(shelf.Request request) async {
       new PackageQuery.fromServiceQueryParameters(request.url.queryParameters));
   return jsonResponse(result.toJson(), indent: indent);
 }
+
+/// Handles requests for:
+///   - /packages/<package>
+Future<shelf.Response> packageHandler(shelf.Request request) async {
+  final String path = request.requestedUri.path.substring('/packages/'.length);
+  final String packageName = path;
+  final String requestMethod = request.method?.toUpperCase();
```

Comment:
There should always be a `request.method`, it's not possible to receive an HTTP request without a method.